lua/cfc_ulx_commands/curse/effects/random_sounds.lua

```diff
@@ -0,0 +1,142 @@
+local EFFECT_NAME = "RandomSounds"
+local POOL_SIZE = 100 -- Smaller pool sizes help reduce lag from loading new sounds on the fly (most notable with footsteps).
+local POOL_BATCH_SIZE = 10 -- Builds up the pool in batches over time, since it requires sounds to be loaded to see if they're good.
+local POOL_BATCH_INTERVAL = 1
+local SOUND_DURATION_MAX = 5
+
+
+-- Create global table
+local EFFECT_NAME_LOWER = string.lower( EFFECT_NAME )
+CFCUlxCurse.EffectGlobals[EFFECT_NAME_LOWER] = CFCUlxCurse.EffectGlobals[EFFECT_NAME_LOWER] or {}
+local globals = CFCUlxCurse.EffectGlobals[EFFECT_NAME_LOWER]
+
+local allSounds = nil
+local allSoundsLength = nil
+local poolSounds = {}
+local poolSoundsLength = nil
+local poolSoundsTargetLength = nil
+local entityMeta = FindMetaTable( "Entity" )
+
+local mathRandom = math.random
+local stringFind = string.find
+local stringSub = string.sub
+local soundDuration = SoundDuration
+
+
+local function getSound()
+    return poolSounds[mathRandom( 1, poolSoundsLength )]
+end
+
+local function getSoundForPool()
+    local attempts = 10
+    local snd = nil
+
+    while attempts > 0 do
+        snd = allSounds[mathRandom( 1, allSoundsLength )]
+
+        -- Not perfect, but covers a lot of cases.
+        local isBad =
+            stringFind( snd, "loop" ) or
+            stringSub( snd, 1, 6 ) == "synth/" or
+            soundDuration( snd ) > SOUND_DURATION_MAX
+
+        if isBad then
+            attempts = attempts - 1
+        else
+            break
+        end
+    end
+
+    return snd
+end
```

Comment:
Essentially the same suggestion as further down. You may at least want to extract your `isGoodSound` function somewhere you can share it
```suggestion
-- Not perfect, but covers a lot of cases.
local funcion isGoodSound( snd )
    if stringFind( snd, "loop" ) then return end
    if stringSub( snd, 1, 6 ) == "synth/" then return end
    if soundDuration( snd ) > SOUND_DURATION_MAX then return end
    
    return true
end
    
local function getSoundForPool()
    for _ = 1, 10 do
        local snd = allSounds[mathRandom( 1, allSoundsLength )]
        
        if isGoodSound( snd ) then
            return snd
        end
    end
end
```